tools/tck-api/src/test/resources/org/opencypher/tools/tck/Foo.feature

```diff
@@ -67,6 +67,22 @@ Feature: Foo
       """
     Then a SyntaxError should be raised at compile time: UnknownFunction
 
+  Scenario: Fail with code
+    Given an empty graph
+    When executing query:
+      """
+      RETURN foo()
+      """
+    Then an error with GQL code fooCode should be raised
+
+  Scenario: Fail with code and message
+    Given an empty graph
+    When executing query:
+      """
+      return foo()
+      """
+    Then an error with GQL code fooCode should be raised with message fooMessage
```

Comment:
Asserting on the complete message can be difficult (it often contains the complete query string, and that might be long and in parameterized scenarios (examples) it is different between runs). So I suggest to make it `...with message containing`.

Or perhaps have syntax like:

```
Then an error with GQL code fooCode should be raised with message containing:
  """Blabla"""
```

Reply:
Or you could have:

```
Then an error with GQL code fooCode should be raised with message matching: /regex/
```

The question is what is actually needed. In the oC scenarios, I would advocate for just checking the code, as the message is something vendor specific.

If we want to check the precise message for the product, than I am not opposing these means. But maybe there is another add this so that the scenarios do not have to be duplicated. 

Maybe the TCK API can be given a mapping of code to message regex (by the  Neo4j TCK adapter, in our case). When evaluating the error step of a scenario the TCK API checks is it has a mapping for the code. If it has none, it does not check the message. If it has one, it test the message too. How about that?

This would allow us to test messages for consistency without unnecessarily detailing the oC TCK with messages.

Reply:
This part sounds like a great ideia. Considering GQL codes have a 1 to 1 mapping to messages (which are parameterized).